react-events/my-app/src/App.tsx

```diff
@@ -0,0 +1,38 @@
+import { useState } from 'react';
+import { Header } from './Header';
+import { ImageContainer } from './ImageContainer';
+import { ImageCaption } from './ImageCaption';
+import { ImageDescription } from './ImageDescription';
+import { ButtonContainer } from './ButtonContainer';
+import './App.css';
+
+const headerText = 'React Image Bank';
+const imageSrc = ['/starry-sky.jpeg', '/cool-kitty.jpg', '/cool-car.jpg'];
+const imageCap = ['A Beautiful Image of Space', 'A Cool Cat', 'A Cool Car'];
+const imageDescrip = [
+  'This is a wonderful sky full of stars!',
+  'This is a picture of a very cool cat!',
+  'This is a picture of a very cool car!',
+];
+const buttonLabel = 'Click for Next Image';
+
+export function App() {
+  const [index, setIndex] = useState(0);
+
+  function handleClick() {
+    setIndex((index + 1) % imageSrc.length);
+  }
+
+  return (
+    <>
+      <Header text={headerText} />
+      <ImageContainer onImageClick={handleClick} src={imageSrc[index]} />
```

Comment:
It doesn't hurt to have these click events here. That way you can flip to the next image by clicking any of the elements. However, the exercise wanted to have just the button perform the next image action.